### PR TITLE
Add warning for write level rights with regards to repo secrets

### LINKED ISSUE #11167: Improve warning for handing out `write` level rights with regards to repo secrets
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions

### What part(s) of the article would you like to see updated?

There's a big warning in the linked page:

> - Use credentials that are minimally scoped:
>   - Make sure the credentials being used within workflows have the least privileges required, and be mindful that any user with write access to your repository has read access to all secrets configured in your repository.

It's really easy to read/skim past this the way the document is structured now.

I propose we add a new heading: `Strongly consider who you give write rights to` and
put the warning "Be mindful that any user with write access to your repository has read access to all secrets configured in your repository." in that section.

I suggest styling the warning message itself as a warning box with a red background, or something similar to really capture the attention of the reader.

### Additional information

_No response_

--- 
Edited by maintainer.
Here is the [content design plan ](https://github.com/github/docs/issues/11167#issuecomment-964605058) by a writer for this issue.

### PR DESCRIPTION
**Pull request by mounilKshah**
### Why:

Closes #11167 

### What's being changed:

Added a warning under the **Using Secrets** section.

### Check off the following:

- [ ] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ethomson** [comment]
This feels like it's reiterating https://github.com/github/docs/pull/15553/files#diff-f1d86f75ef58bf6616cfcd111f4821a88984a396b9178ff20b1af709d8ffa6dfL42.  I wonder if we should make improvements to that paragraph instead of duplicating the idea in a different place.  What do you think?  🤔

**Comment by mounilKshah** [comment]
> This feels like it's reiterating https://github.com/github/docs/pull/15553/files#diff-f1d86f75ef58bf6616cfcd111f4821a88984a396b9178ff20b1af709d8ffa6dfL42. I wonder if we should make improvements to that paragraph instead of duplicating the idea in a different place. What do you think? 🤔

Thank you for the review.
Actually, #11167 asks for addition of a warning, which is why I added it. As this is my first time contributing to a documentation, I wasn't sure what would be the best way to write the warning, hence I repeated what was already mentioned in the section. If there is any better way to reframe the warning, I'll make the changes accordingly.

File: content/actions/security-guides/security-hardening-for-github-actions.md [hunk 1]
```diff
@@ -48,6 +48,12 @@ To help prevent accidental disclosure, {% data variables.product.product_name %}
 - **Consider requiring review for access to secrets**
     - You can use required reviewers to protect environment secrets. A workflow job cannot access environment secrets until approval is granted by a reviewer. For more information about storing secrets in environments or requiring reviews for environments, see "[Encrypted secrets](/actions/reference/encrypted-secrets)" and "[Using environments for deployment](/actions/deployment/using-environments-for-deployment)."
 
+{% warning %}
+
+**Warning**: Any user with write access to your repository has read access to all secrets configured in your repository. Therefore, you need to ensure that the credentials being used within workflows have the least privileges required.
```
Comment (skedwards88) [inline content/actions/security-guides/security-hardening-for-github-actions.md:53]:
```suggestion
**Warning**: Any user with write access to your repository has read access to all secrets configured in your repository. Therefore, you should ensure that the credentials being used within workflows have the least privileges required.
```

**Review by skedwards88 — COMMENTED** [review]
Thanks for this PR! I made one minor wording suggestion that I will go ahead and incorporate, then we'll get this merged down for you.

**Review by skedwards88 — APPROVED** [review]

**Pull request merged**